### PR TITLE
Binance Convert API Endpoint added

### PR DESCRIPTION
**Pull request by jnuernberg**
Moin,

I have added the Binance API Convert Endpoint to the binance-connector-python

**Changes:**
- Added file `binance/spot/convert.py`  --  Contains method my_converts with the `/sapi/v1/convert/tradeFlow` endpoint.
- Modified `binance/spot/__init__.py` -- Added include statement for convert endpoint.
- Added file `examples/spot/convert` -- Demonstrates the usage of the convert api endpoint.

I hope this helps, cheers.

### CODE REVIEW

File: binance/spot/convert.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from binance.lib.utils import check_required_parameters
+
+
+def my_converts(self, startTime, endTime, **kwargs):
```
Comment (aisling11) [inline binance/spot/convert.py:4]: - Please consider method name `convert_trade_history`, being more detailed can provide clear meaning of the method.
- Please also add the method docstring, you can take example from other modules' methods. It's used to generate documentation in https://binance-connector.readthedocs.io/en/stable/.

File: examples/spot/convert/convert.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+spot_client = Spot(key, secret)
+
+# datetime uses POSIX timestamp
+start = int(datetime(2021, 11, 15, 23, 59, 59).timestamp() * 1000)
+end = int(datetime(2021, 12, 7, 23, 59, 59).timestamp() * 1000)
+
+converts = spot_client.my_converts(startTime=start, endTime=end)
+
+print(converts)
```
Comment (aisling11) [inline examples/spot/convert/convert.py:22]: We can use logging.info() instead of print()

**Comment by aisling11** [comment]
Hey @jnuernberg, appreciate you taking the time to contribute to this project. :)

Besides the comments on the code, I would like to also ask if you mind adding other files too in order to complete this new endpoint addition.

- Update CHANGELOD.md with the following and then ` /docs/source/CHANGELOG.rst` as well:
```
## Unreleased

### Added
- New endpoint for Convert:
   - GET /sapi/v1/convert/tradeFlow to support user query convert trade history records
```

- Add a new test module for convert in https://github.com/binance/binance-connector-python/tree/master/tests/spot and create the tests for the method.

- Add a binance.spot.convert.rst file in https://github.com/binance/binance-connector-python/tree/master/docs/source

( We'll create a CONTRIBUTION.md file with guidelines in the future to help. )

File: binance/spot/convert.py (hunk 1 above)
Comment (jnuernberg) [reply binance/spot/convert.py]: Done, detailed answer below.

File: examples/spot/convert/convert.py (hunk 2 above)
Comment (jnuernberg) [reply examples/spot/convert/convert.py]: Done, detailed answer below.

**Comment by jnuernberg** [comment]
Moin @aisling11, thank you very much for your remarks. I wanted to use the converts endpoint but saw, that it was not yet implemented here. So I took the liberty to go ahead and write it :) 

I think I implemented all the changes, that you mentioned in your comments. I would be happy to see them being satisfactory. 

Please read the tests particulary thoroughly for I don't have a programming background and usually do not use automated tests in my code. Therefore, I woud kindly rely on you to verify the correctness of the test implementation.

Best regards

File: tests/spot/convert/test_convert_trade_history.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+
+params = {
+    "startTime": current_timestamp(),
+    "endTime": current_timestamp(),
+}
+
+
+def test_convert_trade_history_without_startTime():
+    """Tests the API endpoint to retreive convert trade history without startTime"""
```
Comment (aisling11) [inline tests/spot/convert/test_convert_trade_history.py:21]: typo, should be "retrieve"

File: tests/spot/convert/test_convert_trade_history.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    params = {"startTime": "", "endTime": "1597130241000"}
```
Comment (aisling11) [inline tests/spot/convert/test_convert_trade_history.py:23]: endTime should be int type

File: tests/spot/convert/test_convert_trade_history.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    client = Client(key, secret)
+    client.convert_trade_history.when.called_with(**params).should.throw(
```
Comment (aisling11) [inline tests/spot/convert/test_convert_trade_history.py:25]: we can directly use `called_with("", 1597130241000)` here

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Added
+- New endpoint for Convert:
+   - GET /sapi/v1/convert/tradeFlow to support user query convert trade history records
```
Comment (aisling11) [inline CHANGELOG.md:7]: sorry, forgot to mention about the right indention and use of quote marks here, please take example from previous additions :)

File: binance/spot/convert.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+    GET /sapi/v1/convert/tradeFlow
+
+    https://binance-docs.github.io/apidocs/spot/en/#convert-endpoints
+
+    Args:
+        startTime (int)
+        endTime (int)
+    Keyword Args:
+        limit (int, optional): Default Value: 500; Max Value: 1000
```
Comment (aisling11) [inline binance/spot/convert.py:17]: description here should be: "Default 100, Max 1000"

**Comment by aisling11** [comment]
Appreciate the the quick update @jnuernberg! Just added few comments :)

In addition, we've created a rc-1.9.0 branch, where this merge should go ("master" is for the final release), could you change the PR to that destination, please?

Also, in case you want to explore more, the tools used for tests files is `pytest` and for code style is `black`.

File: tests/spot/convert/test_convert_trade_history.py (hunk 3 above)
Comment (jnuernberg) [reply tests/spot/convert/test_convert_trade_history.py]: Yes, a clear type... copy pasted 3 times :)

File: tests/spot/convert/test_convert_trade_history.py (hunk 5 above)
Comment (jnuernberg) [reply tests/spot/convert/test_convert_trade_history.py]: I changed it to a function call with a keyword argument. I figure this was clearer, just according to pythons "explicit is better than implicit".

**Comment by jnuernberg** [comment]
Hi @aisling11, Thank you for the fast feedback! I find your comments extremely helpful.

I think I caught everything by now! Let me know, if there are more changes required. Also I started reading up on `pytest`, however, I did not find the time to really dig deep into it. It looks very interesting.

Have a nice weekend!

**Comment by aisling11** [comment]
Thanks @jnuernberg, looks all good now! :)

File: tests/spot/convert/test_convert_trade_history.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    client = Client(key, secret)
+    client.convert_trade_history.when.called_with(startTime=1597130241000).should.throw(
+        ParameterRequiredError
+    )
+
+
+@mock_http_response(
+    responses.GET, "/sapi/v1/convert/tradeFlow\\?" + urlencode(params), mock_item, 200
+)
+def test_futures_transfer_history():
```
Comment (aisling11) [inline tests/spot/convert/test_convert_trade_history.py:41]: Change the method to be "convert_trade_history"

File: tests/spot/convert/test_convert_trade_history.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    "startTime": current_timestamp(),
+    "endTime": current_timestamp(),
+}
+
+
+def test_convert_trade_history_without_startTime():
+    """Tests the API endpoint to retrieve convert trade history without startTime"""
+
+    client = Client(key, secret)
+    client.convert_trade_history.when.called_with(endTime=1597130241000).should.throw(
```
Comment (aisling11) [inline tests/spot/convert/test_convert_trade_history.py:24]: Needs to be `(startTime="",endTime=1597130241000)` -> Same logic for the 2nd test

File: tests/spot/convert/test_convert_trade_history.py (hunk 8 above)
Comment (jnuernberg) [reply tests/spot/convert/test_convert_trade_history.py]: Pardon my oversight. Resolved the copy pase error.

**Comment by aisling11** [comment]
Thanks @jnuernberg! The GH pipeline failed, but it's just  [black](https://pypi.org/project/black/) code style error, which we'll manage in the final version of rc1.9.0 👍

**Comment by jnuernberg** [comment]
> Thanks @jnuernberg! The GH pipeline failed, but it's just [black](https://pypi.org/project/black/) code style error, which we'll manage in the final version of rc1.9.0 👍

Happy to help! And thank you for your remarks and for pointing me towards `pytest` and `black` 👍 . And apologies for the repeated back and forth. As mentioned, I don't have a programmers background.

**Pull request merged**